client/my-sites/site-settings/controller.js

```diff
@@ -116,6 +117,16 @@ export function legacyRedirects( context, next ) {
 	next();
 }
 
+export function redirectToGeneral( context, next ) {
+	const site = getSiteFragment( context.path );
+
+	if ( site ) {
```

Comment:
I don't see how to test this part of the logic. Since it's only added as a middleware to the `'/settings/:section'` route, we'll actually never have a site fragment in the current route, making `redirectToGeneral` essentially useless. Perhaps I misunderstand - could you elaborate on how this is supposed to work?